packages/jaeger-ui/src/components/TracePage/CriticalPath/index.tsx

```diff
@@ -0,0 +1,125 @@
+export const findChildSpanIds = (spans: Span[]): Span[] => {
+  const refinedSpanData: Span[] = [];
+  spans.forEach(span => {
+    if (span.hasChildren) {
+      const Children = spans
+        .filter(span2 =>
+          span2.references.some(
+            reference => reference.refType === 'CHILD_OF' && reference.spanID === span.spanID
+          )
+        )
+        .sort((a, b) => b.startTime + b.duration - (a.startTime + a.duration))
+        .map(span2 => span2.spanID);
+      refinedSpanData.push({ ...span, childSpanIds: Children });
+    } else {
+      refinedSpanData.push({ ...span, childSpanIds: [] });
+    }
+  });
+  return refinedSpanData;
+};
+
+// This function finds LFC of a current span which is less than the spawn event
+export const findLastFinishingChildSpanId = (
+  traceData: Trace,
+  currentSpan: Span,
+  spawnTime?: number
+): string | undefined => {
+  if (spawnTime) {
+    return currentSpan.childSpanIds.find(each =>
+      traceData.spans.some(span => span.spanID === each && span.startTime + span.duration < spawnTime)
+    );
+  }
+  return currentSpan.childSpanIds[0];
+};
+
+export const computeCriticalPath = (
+  traceData: Trace,
+  spanId: string,
+  criticalPath: criticalPathSection[],
+  spawnTime?: number
+) => {
+  // spawnTime refers to spawn synchronization event of current span
+  const currentSpan: Span = traceData.spans.filter(span => span.spanID === spanId)[0];
+
+  const lastFinishingChildSpanId = findLastFinishingChildSpanId(traceData, currentSpan, spawnTime);
+  let spanCriticalSection: criticalPathSection;
+
+  if (lastFinishingChildSpanId) {
+    const lfcSpan = traceData.spans.filter(span => span.spanID === lastFinishingChildSpanId)[0];
+    spanCriticalSection = {
+      spanId: currentSpan.spanID,
+      section_start: lfcSpan.startTime + lfcSpan.duration,
+      section_end: spawnTime || currentSpan.startTime + currentSpan.duration,
+    };
+    criticalPath.push(spanCriticalSection);
+    // Now lfc turns to the currentspan and we again do the same algorithm
+    computeCriticalPath(traceData, lastFinishingChildSpanId, criticalPath);
+  } else {
+    // If there is no last finishing child then total section upto startTime of span is on critical path
+    spanCriticalSection = {
+      spanId: currentSpan.spanID,
+      section_start: currentSpan.startTime,
+      section_end: spawnTime || currentSpan.startTime + currentSpan.duration,
+    };
+    criticalPath.push(spanCriticalSection);
+    // Now as there are no lfc's it goes back to parent span from startTime(spawn Event)
+    if (currentSpan.references.length) {
+      const parentSpan: string = currentSpan.references.filter(
+        reference => reference.refType === 'CHILD_OF'
+      )[0].spanID;
+      computeCriticalPath(traceData, parentSpan, criticalPath, currentSpan.startTime);
+    }
+  }
+  return criticalPath;
+};
+
+function TraceCriticalPath(props: Tprops) {
```

Comment:
Can we rename it to something more concrete? Like props in this case is a traceProp, so let's use that.
```suggestion
function TraceCriticalPath(tracePop: Tprops) {
```